stock_picking_product_generate_sequence/models/stock_move_line.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# All Right Reserved
+#
+# See LICENSE file for full licensing details.
+##############################################################################
+from re import findall as regex_findall
+from re import split as regex_split
+
+from odoo import models, fields, api
+
+
+class StockMoveLine(models.Model):
+    _inherit = 'stock.move.line'
+
+    def action_generate_product_serial(self):
+        self.ensure_one()
+        self.lot_id = self.env['stock.production.lot'].create({
+            'product_id': self.product_id.id,
+            'company_id': self.company_id.id,
+            'name': self._get_next_serial(self.company_id, self.product_id) or
+                    self.env['ir.sequence'].next_by_code('stock.picking.product')
+        })
+
+    @api.model
+    def _get_next_serial(self, company, product):
+        sequence = product.product_tmpl_id.serial_lot
+        if sequence:
+            splitted = regex_split(r'(\d+)', sequence)
+            prefix = splitted[0]
+            initial_number = int(splitted[1])
+            padding = len(splitted[1])
+            last_serial = self.env['stock.production.lot'].search(
+                [
+                    ('company_id', '=', company.id),
+                    ('product_id', '=', product.id),
+                    ('name', '=ilike', '%s%%' % prefix)
```

Comment:
[Question] Why %s%%, is it really necessary 2 %%?

Reply:
because if I put only a % I get an error saying that it is not in the correct format. It would be only 1 % if I was using "=like" instead of "=ilike" because with "like" (as I have read) Odoo itself adds internally the missing %
![Screenshot from 2022-08-10 09-09-53](https://user-images.githubusercontent.com/97018161/184179665-43e3e1d1-d5ba-4e1d-b77b-6b42ff629b68.png)